assets/js/atomic/blocks/product-elements/rating/block.tsx

```diff
@@ -14,20 +13,49 @@ import {
 	useTypographyProps,
 } from '@woocommerce/base-hooks';
 import { withProductDataContext } from '@woocommerce/shared-hocs';
+import { isNumber } from '@woocommerce/types';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
+type Props = {
+	className?: string;
+};
+
+const getAverageRating = (
+	// TODO: This is a workaround until we can confirm the correct type from ProductResponseItem.
```

Comment:
Why can't we confirm the correct type?

Also note that we have an automation that creates new issues with every `TODO` comment. I'm not sure we want to create an issue for this specifically. Maybe we can create the issue that ties into some other TS migration?

Reply:
I did a quick API call to confirm. `average_rating` is indeed a string. This is the response I got:

```
{
	"id": 16,
	"name": "Beanie",
	"parent": 0,
	"type": "simple",
	"variation": "",
	"permalink": "http:\/\/kirigami.local\/product\/beanie\/",
	"sku": "woo-beanie",
	"short_description": "<p>This is a simple product.<\/p>",
	"description": "<p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.<\/p>",
	"on_sale": true,
	"prices": {
		"price": "1800",
		"regular_price": "2000",
		"sale_price": "1800",
		"price_range": null,
		"currency_code": "EUR",
		"currency_symbol": "€",
		"currency_minor_unit": 2,
		"currency_decimal_separator": ",",
		"currency_thousand_separator": ".",
		"currency_prefix": "",
		"currency_suffix": " €"
	},
	"price_html": "<del aria-hidden=\"true\"><span class=\"woocommerce-Price-amount amount\">20,00&nbsp;<span class=\"woocommerce-Price-currencySymbol\">&euro;<\/span><\/span><\/del> <ins><span class=\"woocommerce-Price-amount amount\">18,00&nbsp;<span class=\"woocommerce-Price-currencySymbol\">&euro;<\/span><\/span><\/ins>",
	"average_rating": "5.00",
	"review_count": 1,
	"images": [
		{
			"id": 45,
			"src": "http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2.jpg",
			"thumbnail": "http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-450x450.jpg",
			"srcset": "http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2.jpg 801w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-450x450.jpg 450w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-100x100.jpg 100w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-600x600.jpg 600w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-300x300.jpg 300w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-150x150.jpg 150w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-768x768.jpg 768w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-324x324.jpg 324w, http:\/\/kirigami.local\/wp-content\/uploads\/2022\/01\/beanie-2-416x416.jpg 416w",
			"sizes": "(max-width: 801px) 100vw, 801px",
			"name": "beanie-2.jpg",
			"alt": ""
		}
	],
	"categories": [
		{
			"id": 19,
			"name": "Accessories",
			"slug": "accessories",
			"link": "http:\/\/kirigami.local\/product-category\/clothing\/accessories\/"
		}
	],
	"tags": [],
	"attributes": [
		{
			"id": 1,
			"name": "Color",
			"taxonomy": "pa_color",
			"has_variations": false,
			"terms": [
				{
					"id": 24,
					"name": "Red",
					"slug": "red"
				}
			]
		}
	],
	"variations": [],
	"has_options": false,
	"is_purchasable": true,
	"is_in_stock": true,
	"is_on_backorder": false,
	"low_stock_remaining": null,
	"sold_individually": false,
	"add_to_cart": {
		"text": "Add to cart",
		"description": "Add &ldquo;Beanie&rdquo; to your cart",
		"url": "?add-to-cart=16",
		"minimum": 1,
		"maximum": 9999,
		"multiple_of": 1
	},
	"extensions": {}
}
```

So, my suggestion is to fix the wrong interface. I did a quick search on our repo, and it seems that the `average_rating` as a number is used in only two cases, so that could also be fixed relatively quickly.

If you still believe to be best not to fix the interface right now, you can do this:

```
product: Omit< ProductResponseItem, 'average_rating' > & {
  average_rating: string;
}
```

Reply:
Thanks for the info here, @sunyatasattva, this is very helpful!

> it seems that the `average_rating` as a number is used in only two cases, so that could also be fixed relatively quickly.

Yeah, I am seeing the same thing - showing up in previews (`assets/js/previews/cart.ts` and `assets/js/previews/product.js`). While it should be a relatively quick fix, I think I'd prefer to address that separately as a follow-up, if that's ok. That way I can keep this work moving forward and scoped to the task. We can also test to confirm there's no impact on updating that interface without holding things up here.

Reply:
I've created a new issue to address the incorrect type on the `ProductResponseItem` at #7408, and assigned myself to it.

I've also updated the type-casting here, per your suggestion, and removed the `TODO` comments in 9eab7cbc7.